### PR TITLE
Select region by label

### PR DESCRIPTION
**Pull request by f-fl0**
This PR provides a way to select a region of point sharing the label (to delete them for example). Here is a video showing this feature in action. A part of the map cloud is labeled (blue points). The user `Ctrl+Left click` one of them and the surrounding region of blue points is automatically selected. These points can then be easily deleted for example.

https://user-images.githubusercontent.com/66578286/138251576-7d4c9366-4611-4014-8283-3a14f43d09a9.mp4

### CODE REVIEW

File: command.go [hunk 1]
```diff
@@ -896,3 +913,63 @@ func (c *commandContext) SelectSegment(p mat.Vec3) {
...
+				v.AddByAddr(a, i)
+			}
+		}
+		it.Incr()
+	}
+	if it, err = c.editor.pp.Vec3Iterator(); err != nil {
+		return err
+	}
+
+	vIndice := v.Storage().Indice()
```
Comment (at-wat) [inline command.go:947]: It's not needed to be stored to VoxelGrid storage.
```suggestion
	vIndice := make([]int, 0, ??)
	n := c.editor.pp.Points
	for i := 0; i < n; i++ {
		c.selectMask[i] &= ^uint32(selectBitmaskSegmentSelected)
		if c.selectMask[i]&(selectBitmaskCropped|selectBitmaskOnScreen) == selectBitmaskOnScreen {
			continue
		}
		v := it.Vec3At(i).Sub(p)
		if v[0] < -c.segmentationRange || c.segmentationRange < v[0] ||
			v[1] < -c.segmentationRange || c.segmentationRange < v[1] ||
			v[2] < -c.segmentationRange || c.segmentationRange < v[2] {
			continue
		}
		vIndice = append(vIndice, i)
	}
```

File: go.sum [hunk 2]
```diff
@@ -1,5 +1,7 @@
 github.com/seqsense/pcgol v0.0.0-20211018090702-f81f4d78c09f h1:SmuZvR5cGJa8IjcdRVfUoRZ8uq0CrZ9OG3zjYezQNbc=
 github.com/seqsense/pcgol v0.0.0-20211018090702-f81f4d78c09f/go.mod h1:w8LffJCFuzmJsqCRLWNB2SXOVHlEPtHBCHmlisG2w+0=
```
Comment (at-wat) [inline go.sum:2]: Seems not go mod tidied

File: README.md [hunk 3]
```diff
@@ -21,6 +21,7 @@ Shift + 左クリック   | 2番目、4点目の点選択 [\*1](#footnote1)
 中ドラッグ           | 視点移動
 Shift + 左ドラッグ   | 視点移動
 Alt + 左クリック     | 隣接する点群を選択 [\*2](#footnote2)
+Ctrl + 左クリック    | ラベルを元にして隣接する点群を選択
```
Comment (at-wat) [inline README.md:24]: 同じラベルの隣接する点群を選択 maybe better

**Comment by f-fl0** [comment]
CI failed on an unrelated test after b7f29553a7380f326fdd323f4db3f7e0156bf80c.
```
=== RUN   TestWheelNormalizer/AnalogWheel30
--- FAIL: TestWheelNormalizer (0.36s)
    --- PASS: TestWheelNormalizer/BinaryWheel1 (0.09s)
    --- PASS: TestWheelNormalizer/BinaryWheel10 (0.09s)
    --- FAIL: TestWheelNormalizer/AnalogWheel3 (0.09s)
    --- PASS: TestWheelNormalizer/AnalogWheel30 (0.09s)
FAIL
coverage: 34.9% of statements in ./...
```

File: README.md (hunk 3 above)
Comment (f-fl0) [reply README.md]: Done in 7bdaddd1c4eb0a3ef1a4905e8f0cced06733439b

File: command.go (hunk 1 above)
Comment (f-fl0) [reply command.go]: I made some adjustments to your suggestion in 376d67eb4165aba9cad35d561fcf906649b6845f.

File: command.go [hunk 4]
```diff
@@ -896,3 +917,57 @@ func (c *commandContext) SelectSegment(p mat.Vec3) {
...
+	searchDistance := c.labelSegmentationSearchDistance
+	kdt := kdtree.New(raIn)
+	nn := kdt.Nearest(p, searchDistance)
+	if nn.ID < 0 {
+		return fmt.Errorf("no point close to %v", p)
+	}
+
+	l := raLIn.Uint32At(nn.ID)
+	if l == 0 {
+		return fmt.Errorf("%v is not labeled", p)
```
Comment (at-wat) [inline command.go:958]: Isn't there a case user want to select label=0 points?

**Review by at-wat — APPROVED** [review]
LGTM

File: command.go (hunk 4 above)
Comment (f-fl0) [reply command.go:958]: I guess you're right. I'll take that condition out.

Comment (f-fl0) [reply command.go]: Done in 9ac3ce813aa2038819c9ae96ec0ac8bbece4ecb6

**Pull request merged**